lib/Test.pm6

```diff
@@ -611,11 +611,21 @@ sub throws-like($code, $ex_type, $reason?, *%matcher) is export {
                 if $type_ok {
                     for %matcher.kv -> $k, $v {
                         my $got is default(Nil) = $_."$k"();
-                        my $ok = $got ~~ $v,;
-                        ok $ok, ".$k matches $v.gist()";
-                        unless $ok {
-                            _diag "Expected: " ~ ($v ~~ Str ?? $v !! $v.perl)
-                              ~ "\nGot:      $got";
+                        # Match strings and Regex as regex, everything else gets a direct comparison.
+                        if (nqp::eqaddr($v.WHAT, Str) or nqp::eqaddr($v.WHAT, Regex)) {
+                            my $ok = $got ~~ $v,;
```

Comment:
The trailing comma was in the original code, but looks like it was just a typo there, maybe get rid of it while you're making changes?